pymc_experimental/inference/find_map.py

```diff
@@ -0,0 +1,431 @@
+import logging
+
+from collections.abc import Callable
+from typing import Literal, cast, get_args
+
+import jax
+import numpy as np
+import pymc as pm
+import pytensor
+import pytensor.tensor as pt
+
+from better_optimize import minimize
+from better_optimize.constants import MINIMIZE_MODE_KWARGS, minimize_method
+from pymc.blocking import DictToArrayBijection, RaveledVars
+from pymc.initial_point import make_initial_point_fn
+from pymc.model.transform.optimization import freeze_dims_and_data
+from pymc.pytensorf import join_nonshared_inputs
+from pymc.util import get_default_varnames
+from pytensor.compile import Function
+from pytensor.compile.mode import Mode
+from pytensor.tensor import TensorVariable
+from scipy.optimize import OptimizeResult
+
+_log = logging.getLogger(__name__)
+
+GradientBackend = Literal["pytensor", "jax"]
+VALID_BACKENDS = get_args(GradientBackend)
+
+
+def set_optimizer_function_defaults(method, use_grad, use_hess, use_hessp):
+    method_info = MINIMIZE_MODE_KWARGS[method].copy()
+
+    use_grad = use_grad if use_grad is not None else method_info["uses_grad"]
+    use_hess = use_hess if use_hess is not None else method_info["uses_hess"]
+    use_hessp = use_hessp if use_hessp is not None else method_info["uses_hessp"]
+
+    if use_hess and use_hessp:
```

Comment:
I was going through all the methods thinking when you would need `hess` and `hessp` and then came back to this. I would probably warn the user / not let them pass _both_ `use_hess` and `use_hessp`